functions/snippets/src/main/java/HelloHttpSample.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_helloworld_http]
+
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.logging.Logger;
+import org.apache.commons.io.IOUtils;
+
+public class HelloHttpSample implements HttpFunction {
+  private static final Logger LOGGER = Logger.getLogger(HelloHttpSample.class.getName());
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+    String name = "world";
+
+    try {
+      String requestString = IOUtils.toString(request.getReader());
```

Comment:
We should be consistent and use GSON here ([example](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/cad876ad205e03a19abafcd16083b2790d84db95/functions/snippets/src/main/java/ParseContentType.java#L30))

It's also good practice to verify the content type and send back an error response if the input isn't valid.

Reply:
Also, a comment for this section would be helpful. Something like "Parse request body as a JSON to check 'name' field" so it's clear that's what's being attempted.